chargebee/src/main/java/com/chargebee/android/billingservice/GPErrorCode.kt

```diff
@@ -18,4 +23,84 @@ enum class GPErrorCode(val errorMsg: String) {
     DeveloperError("Invalid arguments provided to the API"),
     BillingClientNotReady("Play services not available"),
     SDKKeyNotAvailable("SDK key not available to proceed purchase"),
+    InvalidPurchaseToken("The Token data sent is not correct or Google service is temporarily down"),
+}
+
+internal enum class BillingErrorCode(val code: Int, val message: String) {
+    UNKNOWN(-4, "Unknown error occurred"),
+    SERVICE_TIMEOUT(-3, "The request has reached the maximum timeout before Google Play responds"),
+    FEATURE_NOT_SUPPORTED(
+        -2,
+        "The requested feature is not supported by the Play Store on the current device"
+    ),
+    USER_CANCELED(1, "Transaction was canceled by the user"),
+    SERVICE_UNAVAILABLE(2, "The service is currently unavailable"),
+    BILLING_UNAVAILABLE(3, "A user billing error occurred during processing"),
+    ITEM_UNAVAILABLE(4, "The requested product is not available for purchase"),
+    DEVELOPER_ERROR(5, "Error resulting from incorrect usage of the API"),
+    ERROR(6, "Fatal error during the API action"),
+    ITEM_NOT_OWNED(8, "Requested action on the item failed since it is not owned by the user"),
+    SERVICE_DISCONNECTED(
+        -1,
+        "The app is not connected to the Play Store service via the Google Play Billing Library"
+    ),
+    ITEM_ALREADY_OWNED(7, "The purchase failed because the item is already owned");
```

Comment:
Can we double check that there are no changes to the current error message/codes for all scenarios?

Reply:
Also lets document the scenarios and the error codes returned from SDK for the same.